src/sidebars/favourites/tearout.js

```diff
@@ -11,37 +11,53 @@
                         var dragElement = element[0],
                             tearElement = dragElement.parentNode.parentNode,
                             tileWidth = tearElement.clientWidth,
-                            tileHeight = tearElement.clientHeight;
+                            tileHeight = tearElement.clientHeight,
+                            store;
 
-                        function createConfig() {
+                        function createConfig(minWidth, minHeight, width, height, url, resizable, maximizable, showTaskbarIcon, saveWindowState) {
```

Comment:
I'm a little late to the party, but here's my 2 cents: I'd say that a json object makes more sense, but considering that this function essentially just spits out what it takes in, but formatted into an object; is this function even necessary at all? This would be an easier decision if we were using typescript where we could enforce the format of the json object a bit better, but still, I'm inclined to say that a json object is better than tons of params. I see your point @owennw about the lack of defaults and accidentally missing things, but as it is you might accidentally miss something whilst putting the parameters in, or you might get them in the wrong order, which could lead to weird behaviour. However since it is only called in two places, it's not that bad I guess